packages/ckeditor5-html-support/src/datafilter.js

```diff
@@ -288,6 +301,83 @@ export default class DataFilter extends Plugin {
 		}, { priority: 'lowest' } );
 	}
 
+	/**
+	 * Registers a model post-fixer that is removing coupled GHS attributes of inline elements. Those attributes
+	 * are removed if a coupled feature attribute is removed.
+	 *
+	 * For example while unlinking a text (by the remove of `linkHref` attribute):
+	 *
+	 *		<$text linkHref="foo" htmlA="{ ... }">bar</$text>
+	 *
+	 * The `htmlA` attribute would be left behind in the model and would cause GHS to generate an `<a>` element
```

Comment:
```suggestion
	 * For example, consider following HTML:
	 *
	 *		<a href="foo.html" id="myId"></a>
	 *
	 * Which would be upcasted to following text node in the model:
	 *
	 *		<$text linkHref="foo.html" htmlA="{ id: 'myId' }">bar</$text>
	 *
	 * When the user removes the link from that text (using UI), only `linkHref` attribute would be removed:
	 *
	 *		<$text htmlA="{ id: 'myId' }">bar</$text>
	 *
	 * The `htmlA` attribute would stay in the model and would cause GHS to generate an `<a>` element.
	 * This is incorrect from UX point of view, as the user wanted to remove the whole link.
```

Reply:
More verbose example is needed to understand while this mechanism is necessary.

I am not sure if `htmlA` value is correct in the proposed example.